cam.go

```diff
@@ -371,19 +416,22 @@ func (fc *filteredCamera) images(ctx context.Context, extra map[string]interface
 		if shouldSend {
 			// this updates the CaptureTill time to be further in the future
 			fc.buf.MarkShouldSend(meta.CapturedAt)
-			fc.buf.CacheImages(images)
 
-			if x, ok := fc.buf.PopFirstToSend(); ok {
-				return x.Imgs, x.Meta, nil
+			if bufferedImages, bufferedMeta, ok := fc.getBufferedImages(singleImageMode); ok {
+				return bufferedImages, bufferedMeta, nil
 			}
 
-			return images, meta, nil
+			// Don't return the trigger image to maintain chronological order
+			// Represents the edge case where triggering is happening faster than the buffer can be filled
+			return nil, meta, data.ErrNoCaptureToStore
 		}
 	}
-	// background loop will fill ToSend buffer as long as within CaptureTill time
-	if x, ok := fc.buf.PopFirstToSend(); ok {
-		return x.Imgs, x.Meta, nil
+	// No triggers met and we're outside capture window, but check if we have buffered images from previous triggers
+	if bufferedImages, bufferedMeta, ok := fc.getBufferedImages(singleImageMode); ok {
```

Comment:
Not convinced that we need this one but I may be wrong.  Shouldn't the images have been sent already?

Reply:
This is much much more obvious when data capture is slow, but you're using ReadImage and can only pop 1 image off the buffer at a time. The example is:
1. data capture only requests an image every 5 seconds for some reason (something is slowing it down)
2. ToSend buffer has been filled with data every second, and the capture window was long (90s)
3. So you can only work through your 90 image long ToSend buffer in 5 second increments (because that's how fast ReadImage is pinging the filtered-camera)
4. So Even though the capture window has ended long ago, and there are no triggers, the buffer is still popping off images.
5. This wouldn't be the case for Images(), where it can pop all the images off at the same time